src/Health/Checks/ElasticsearchHealthCheck.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\Health\Checks;
+
+use EoneoPay\Externals\DataTransferObjects\Health\HealthState;
+use EoneoPay\Externals\Health\Interfaces\HealthCheckInterface;
+use EoneoPay\Externals\Health\Interfaces\HealthInterface;
+use LoyaltyCorp\Search\Interfaces\ClientInterface;
+
+class ElasticsearchHealthCheck implements HealthCheckInterface
+{
+    /**
+     * The search client instance.
+     *
+     * @var \LoyaltyCorp\Search\Client
+     */
+    private $client;
+
+    /**
+     * Constructs a new instance of the Elasticsearch health check.
+     *
+     * @param \LoyaltyCorp\Search\Interfaces\ClientInterface $client
+     */
+    public function __construct(
+        ClientInterface $client
+    ) {
+        $this->client = $client;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function check(): HealthState
+    {
+        try {
+            $this->client->getIndices();
```

Comment:
Lets move to using this API:

https://www.elastic.co/guide/en/elasticsearch/reference/6.8/cluster-stats.html

You should check that the "status" key is yellow or green. If no response or red, degraded

Reply:
There is a PR for this implementation here: https://github.com/eonx-com/search/pull/28